tv/tv.cpp

```diff
@@ -131,37 +144,45 @@ struct TVLegacyPass final : public llvm::ModulePass {
       return false;
     }
 
+    if (first || skip_verify) {
+      I->second.fn = move(*fn);
+      if (!opt_always_verify)
+        // Prepare syntactic check
+        I->second.fn_tostr = toString(I->second.fn);
+      printDot(I->second.fn, I->second.n++);
+      return false;
+    }
+
     Transform t;
     t.src = move(I->second.fn);
     t.tgt = move(*fn);
 
-    if (opt_print_dot) {
-      string prefix = to_string(I->second.n++);
-      t.tgt.writeDot(prefix.c_str());
-    }
+    bool regenerate_tgt = verify(t, I->second);
+
+    if (regenerate_tgt)
+      I->second.fn = *llvm2alive(F, *TLI);
+    else
+      I->second.fn = move(t.tgt);
+
+    I->second.fn_tostr = toString(I->second.fn);
```

Comment:
should we move this to the regenerate_tgt branch?

Reply:
I think it should be here - `I.second.fn_tostr` was containing src's syntax regardless of regenerate_tgt.

Reply:
Regarding this issue, I'm not sure whether the original code wasn't fully correct.. `fn_tostr` must have been refreshed, but it was missing.

Reply:
agreed the previous code wasn't good. But if we don't regenerate the code, then we don't need to regen the string either.

Reply:
I slightly tweaked the code to keep the previous string if regenerate_tgt is false.